src/hotspot/os/aix/loadlib_aix.cpp

```diff
@@ -304,6 +303,35 @@ static bool reload_table() {
 
 } // end LoadedLibraries::reload()
 
+// Callback for loaded module information (from os.hpp)
+// Input parameters:
+//    char*     module_file_name,
+//    address   module_base_addr,
+//    address   module_top_addr,
+//    void*     param
+static bool for_each_internal(os::LoadedModulesCallbackFunc cb, void* param) {
+
+  for (const loaded_module_t* lm = g_first; lm; lm = lm->next) {
+    (*cb)(lm->shortname,
+          (address) lm->text,
+          (address) lm->text + lm->text_len,
+          param);
+  }
+
+  return true;
+}
+
+// Helper for copying loaded module info to external callers.
+// To avoid dangling pointers 'next' is set to null
+static bool copy_lm_to_external(const loaded_module_t* const from, loaded_module_t* to) {
```

Comment:
So, just to understand, the point of this function is to handle to==NULL (from is never NULL) and to set next to NULL? 

I'd probably keep doing this inline in the two places where this happens, but I leave it up to you. If you think this is cleaner, its fine too.

Reply:
The purpose was actually to work through a compiler error that cropped up when I changed loaded_module_t. It ended up being simple to fix, but I left the plumbing in. I agree this is not worth keeping around. Thanks for pointing this out.